include/factory_resolver.h

```diff
@@ -47,11 +47,11 @@ namespace reinforcement_learning  {
    */
   using time_provider_factory_t = utility::object_factory<i_time_provider, const utility::configuration&>;
 
-  extern data_transport_factory_t& data_transport_factory;
-  extern model_factory_t& model_factory;
-  extern sender_factory_t& sender_factory;
-  extern trace_logger_factory_t& trace_logger_factory;
-  extern time_provider_factory_t& time_provider_factory;
+  __declspec( dllimport ) extern data_transport_factory_t& data_transport_factory;
```

Comment:
This page explains how to handle visibility. https://gcc.gnu.org/wiki/Visibility

Please use our copy of this pattern from the common lib: https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/common/include/vw/common/vwvis.h

Reply:
added vwvis.h to the nuget

Reply:
You should be marking functions exported from the DLL like is done in this file: https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/bindings/c/include/vw/experimental/workspace.h

Then when building the DLL VW_DLL_EXPORTS should be defined seen here: https://github.com/VowpalWabbit/vowpal_wabbit/blob/334d1281dc341aa4f27f4d47b922b6afd1ca1835/bindings/c/CMakeLists.txt#L27